pkg/config/nodetreemodel/config.go

```diff
@@ -332,6 +338,21 @@ func (c *ntmConfig) UnsetForSource(key string, source model.Source) {
 
 	// Replace the child with the node from the previous layer
 	parentNode.InsertChildNode(childName, prevNode.Clone())
+
+	newValue := c.leafAtPathFromNode(key, c.root).Get()
+
+	// Value has not changed, do not notify
+	if reflect.DeepEqual(previousValue, newValue) {
+		return
+	}
+
+	c.sequenceID++
+	receivers := slices.Clone(c.notificationReceivers)
```

Comment:
We don't need to clone the receivers any more since we're holding the lock until the end of the function.